Use OAuthenticator from upstream

GeorgianaElena opened the pull request:
Closes https://github.com/2i2c-org/infrastructure/issues/1252

https://github.com/jupyterhub/oauthenticator/pull/498 has been merged and although there isn't a new release for `oauthenticator`, until then we can still use the upstream repo.

yuvipanda commented:
Thanks for working on this, @GeorgianaElena! I think you need to run `chartpress --push` from inside the `helm-charts` directory to build the image and update the tag.

GeorgianaElena commented:
oops! Thanks @yuvipanda! I've pushed the image now and updated the tag.

Chartpress also updated the basehub version in https://github.com/2i2c-org/infrastructure/blob/master/helm-charts/basehub/Chart.yaml#L7, but that didn't seem to be using the previous tag.

I don't remember if this was intended or it is an error (I feel there was a discussion about his at some point)?

yuvipanda reviewed (approved):
This LGTM, @GeorgianaElena! chartpress updates the basehub version, but I think we should discard that - otherwise, it may cause problems with the daskhub dependency mechanism.

Doesn't have to block this PR, but can you document how to rebuild the hub image too?

GeorgianaElena commented on docs/topic/hub-image.md:
```diff
@@ -0,0 +1,59 @@
+
+When this hub image needs to be updated, the steps to take are:
+
+1. Update the [Dockerfile]((https://github.com/2i2c-org/infrastructure/blob/HEAD/helm-charts/images/hub/Dockerfile) with any changes wanted
```
```suggestion
1. Update the [Dockerfile](https://github.com/2i2c-org/infrastructure/blob/HEAD/helm-charts/images/hub/Dockerfile) with any changes wanted
```

GeorgianaElena commented:
Thanks @yuvipanda! I've just added some docs and I'll merge the PR now 🚀

choldgraf commented on docs/topic/hub-image.md:
````diff
@@ -0,0 +1,59 @@
+
+   - Run chartpress to build the image, push it to the registry and update the basehub helm chart to use the updated image tag
+
+     ```bash
````
extra kudos to @GeorgianaElena for documenting these :-)

damianavila commented on helm-charts/images/hub/Dockerfile:
```diff
@@ -19,7 +19,7 @@ RUN pip install --no-cache git+https://github.com/yuvipanda/jupyterhub-configura
 
 # Brings in https://github.com/jupyterhub/oauthenticator/pull/498
-RUN pip install --no-cache --upgrade git+https://github.com/yuvipanda/oauthenticator@7f3fdc0a14d06f1a081c23b1ceb7060a940d11f8
+RUN pip install --no-cache --upgrade git+https://github.com/jupyterhub/oauthenticator@d189037fe6b57453a3ae733a60740c50f241e713
```
@GeorgianaElena, can we create a follow-up issue to use a released version when that one is released? I have been bitten before when I used unreleased versions...
